x-pack/plugins/cases/public/components/case_form_fields/connector.tsx

```diff
@@ -49,6 +55,10 @@ const ConnectorComponent: React.FC<Props> = ({ connectors, isLoading, isLoadingC
     connectors,
   });
 
+  useEffect(() => {
+    setFieldValue('connectorId', configurationConnector.id);
+  }, [configurationConnector.id, setFieldValue]);
```

Comment:
Unfortunately, I could not find a way to make it work without `useEffect `. This is needed to fix a bug where selecting a solution from the create case flyout would not respect the solutions' configuration.